Consolidate duplicate files (#24320)
dotnet/docs: docs/standard/language-independence.md

To design a component that's language independent, you only need to apply the rules for CLS compliance to your component's public interface. Your private implementation does not have to conform to the specification.

> [!IMPORTANT]
> The rules for CLS compliance apply only to a component's public interface, not to its private implementation.

For example, unsigned integers other than <xref:System.Byte> are not CLS-compliant. Because the `Person` class in the following example exposes an `Age` property of type <xref:System.UInt16>, the following code displays a compiler warning.

[!code-csharp[Conceptual.CLSCompliant#1](../../samples/snippets/csharp/VS_Snippets_CLR/conceptual.clscompliant/cs/public1.cs#1)]
[!code-vb[Conceptual.CLSCompliant#1](../../samples/snippets/visualbasic/VS_Snippets_CLR/conceptual.clscompliant/vb/public1.vb#1)]

> You can make the `Person` class CLS-compliant by changing the type of `Age` property from <xref:System.UInt16> to <xref:System.Int16>, which is a CLS-compliant, 16-bit signed integer. You do not have to change the type of the private `personAge` field.

Reviewer: ```suggestion
You can make the `Person` class CLS-compliant by changing the type of the `Age` property from <xref:System.UInt16> to <xref:System.Int16>, which is a CLS-compliant, 16-bit signed integer. You don't have to change the type of the private `personAge` field.
```

[UInt64](xref:System.UInt64) | 64-bit unsigned integer | [Int64](xref:System.Int64) (may overflow), [BigInteger](xref:System.Numerics.BigInteger), or [Double](xref:System.Double)
[UIntPtr](xref:System.UIntPtr) | Unsigned pointer or handle | [IntPtr](xref:System.IntPtr)

The .NET Class Library or any other class library may include other types that aren't CLS-compliant, for example:

- Boxed value types. The following C# example creates a class that has a public property of type `int*` named `Value`. Because an `int*` is a boxed value type, the compiler flags it as non-CLS-compliant.

  [!code-csharp[Conceptual.CLSCompliant#26](../../samples/snippets/csharp/VS_Snippets_CLR/conceptual.clscompliant/cs/box2.cs#26)]

- Typed references, which are special constructs that contain a reference to an object and a reference to a type. Typed references are represented in .NET by the <xref:System.TypedReference> class.

If a type is not CLS-compliant, you should apply the <xref:System.CLSCompliantAttribute> attribute with an `isCompliant` value of `false` to it. For more information, see [The CLSCompliantAttribute attribute](#the-clscompliantattribute-attribute) section.

> The following example illustrates the problem of CLS compliance in a method signature and in generic type instantiation. It defines an `InvoiceItem` class with a property of type <xref:System.UInt32>, a property of type `Nullable(Of UInt32)`, and a constructor with parameters of type <xref:System.UInt32> and `Nullable(Of UInt32)`. You get four compiler warnings when you try to compile this example.

Reviewer: Is there a reason for using VB syntax here?

Author: Good catch, not sure why it was written that way.
